crucible-debug/src/Lang/Crucible/Pretty.hs

```diff
@@ -106,8 +106,10 @@ ppRegVal iFns tp (C.RV v) =
     C.FunctionHandleRepr args ret -> ppFnVal args ret v
     C.MaybeRepr tpr -> case v of
       W4.Unassigned -> "Nothing"
-      W4.PE cond v' ->
-        "Just" PP.<+> ppRegVal iFns tpr (C.RV v') PP.<+> "if" PP.<+> W4.printSymExpr cond
+      W4.PE cond v'
+        | Just True <- W4.asConstantPred cond -> "Just" PP.<+> ppRegVal iFns tpr (C.RV v')
+        | otherwise ->
+          "Just" PP.<+> ppRegVal iFns tpr (C.RV v') PP.<+> "if" PP.<+> W4.printSymExpr cond
```

Comment:
To avoid duplicating some code, how about writing this like:

```hs
      W4.PE cond v' ->
          let ppCond | Just True <- W4.asConstantPred cond ->
                       "if" PP.<+> W4.printSymExpr cond
                     | otherwise ->
                       PP.empty in
          "Just" PP.<+> ppRegVal iFns tpr (C.RV v') PP.<+> ppCond
```

Reply:
Deduplicated in da984da77b4491d68651920ecbd2b27b9c9d8250